src/Lecture3.hs

```diff
@@ -122,19 +133,24 @@ data Reward = Reward
     } deriving (Show, Eq)
 
 instance Semigroup Reward where
+  (<>) (Reward a True) (Reward b _) = Reward (a <> b) True
+  (<>) (Reward a _) (Reward b True) = Reward (a <> b) True
+  (<>) (Reward a _) (Reward b _) = Reward (a <> b) False
```

Comment:
You don't need to have three cases if you notice that this is the boolean `||` operation